plugin/dashboard.lua

```diff
@@ -33,6 +34,21 @@ api.nvim_create_autocmd('FileType', {
   end,
 })
 
+if db.session_auto_save_on_exit then
+  local session_auto_save = api.nvim_create_augroup('session_auto_save', { clear = true })
+
+  api.nvim_create_autocmd('VimLeavePre', {
+    group = session_auto_save,
+    callback = function()
+      if db_session.session_exists() and vim.fn.len(vim.fn.getbufinfo({ buflisted = 1 })) > 1 then
```

Comment:
`vim.fn.len(vim.fn.getbufinfo({ buflisted = 1 }))` this line for check there has a buffer right ?

Reply:
exactly :+1:

Reply:
```lua
local bufname = api.nvim_buf_get_name(0)
if bufname then ...
```

Reply:
> ```lua
> local bufname = api.nvim_buf_get_name(0)
> if bufname then ...
> ```

This doesn't work. If you open a directory that has a session, don't load it and close out of nvim, the session will be overidden and next time you want to load your session you will only have an empty buffer.

Reply:
I need think about this.

Reply:
> If you open a directory that has a session, don't load it and close out of nvim,

when you do this. the current buffer is `scrath-buffer` .  if it's a buffer without the bufname just return do nothing. it will not overwrite the session.